cmd/dockerd/daemon.go

```diff
@@ -325,7 +325,11 @@ func (cli *DaemonCli) reloadConfig() {
 	}
 }
 
-func (cli *DaemonCli) stop() {
+func (cli *DaemonCli) stop(c chan bool) {
+	if cli.api == nil {
```

Comment:
I don't think this is the right fix as this is only caused by the API not being set on the `cli` struct yet, but it is nonetheless created and needs to be shut down.
I think we should set the api field immediately rather than checking for a nil here.